src/wallet/components/WalletIslandTransactionActions.tsx

```diff
@@ -0,0 +1,113 @@
+import { useOnchainKit } from '@/core-react/useOnchainKit';
+import { addSvgForeground } from '@/internal/svg/addForegroundSvg';
+import { arrowUpRightSvg } from '@/internal/svg/arrowUpRightSvg';
+import { toggleSvg } from '@/internal/svg/toggleSvg';
+import { border, cn, color, pressable, text } from '@/styles/theme';
+import { useCallback } from 'react';
+import { useWalletIslandContext } from './WalletIslandProvider';
+import { useWalletContext } from './WalletProvider';
+
+type TransactionActionProps = {
+  icon: React.ReactNode;
+  label: string;
+  action: () => void;
+};
+
+export function WalletIslandTransactionActions() {
+  const { address } = useWalletContext();
+  const { chain, projectId } = useOnchainKit();
+  const { isFetchingPortfolioData, setShowSwap, animations } =
+    useWalletIslandContext();
+
+  const handleBuy = useCallback(() => {
+    const baseUrl = 'https://pay.coinbase.com/buy/select-asset';
+    const params = [
+      `appId=${projectId}`,
+      `destinationWallets=[{"address": "${address}", "blockchains":["${chain.name.toLowerCase()}"]}]`,
+      'defaultAsset=USDC',
+      'defaultPaymentMethod=CRYPTO_ACCOUNT',
+      'presetFiatAmount=25',
+    ];
+
+    window.open(
+      `${baseUrl}?${params.join('&')}`,
+      'popup',
+      'width=400,height=600,scrollbars=yes',
+    );
+  }, [address, chain.name, projectId]);
```

Comment:
I'd suggest doing it like this - it makes it easier to read because it's more declarative, plus it handles the URL encoding for you.

```suggestion
 const handleBuy = useCallback(() => {
  const url = new URL('https://pay.coinbase.com/buy/select-asset');
  const params = new URLSearchParams({
    appId: projectId,
    destinationWallets: JSON.stringify([{
      address,
      blockchains: [chain.name.toLowerCase()]
    }]),
    defaultAsset: 'USDC',
    defaultPaymentMethod: 'CRYPTO_ACCOUNT',
    presetFiatAmount: '25'
  });

  window.open(`${url}?${params}`, 'popup', 'width=400,height=600,scrollbars=yes');
}, [address, chain.name, projectId]);
```

Reply:
nice, i like that, thank you